docs/src/modules/components/HighlightedCodeWithTabs.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { styled } from '@mui/material/styles';
+import Tabs, { TabsOwnProps } from '@mui/base/Tabs';
+import TabsList from '@mui/base/TabsList';
+import TabPanel from '@mui/base/TabPanel';
+import Tab from '@mui/base/Tab';
+import HighlightedCode from './HighlightedCode';
+
+const StyledTabList = styled('div')(({ theme }) => ({
+  display: 'flex',
+  backgroundColor: (theme.vars || theme).palette.primaryDark[600],
+  borderTopLeftRadius: (theme.vars || theme).shape.borderRadius,
+  borderTopRightRadius: (theme.vars || theme).shape.borderRadius,
+  ...theme.applyDarkStyles({
+    backgroundColor: (theme.vars || theme).palette.primaryDark[700],
+  }),
+}));
+
+const StyledTabPanel = styled('div')<{ ownerState: { mounted: boolean } }>(({ ownerState }) => ({
+  '& pre': {
+    marginTop: 0,
+    borderTopLeftRadius: 0,
+    borderTopRightRadius: 0,
+    '& code': {
+      opacity: ownerState.mounted ? 1 : 0,
+    },
+  },
+}));
+
+const StyledTab = styled('button')<{ ownerState: { mounted: boolean } }>(({ theme, ownerState }) =>
+  theme.unstable_sx({
+    py: 1.5,
+    px: 2,
+    border: 'none',
+    borderBottom: '2px solid transparent',
+    bgcolor: 'primaryDark.800',
+    color: 'rgba(255 255 255 / 0.6)',
+    fontSize: '0.75rem',
+    fontWeight: 600,
+    fontFamily: theme.typography.fontFamilyCode,
+    outline: 'none',
+    minWidth: 80,
+    '&:first-child /* emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason */':
+      {
+        borderTopLeftRadius: (theme.vars || theme).shape.borderRadius,
+      },
+    '&:not(:first-child)': {
+      marginLeft: '1px',
+    },
+    '&:last-child': {
+      borderTopRightRadius: (theme.vars || theme).shape.borderRadius,
+    },
+    ...(ownerState.mounted && {
+      '&[aria-selected="true"]': {
+        color: '#fff',
+        borderColor: (theme.vars || theme).palette.primary.light,
+      },
+    }),
+    '&:hover': {
+      backgroundColor: 'rgba(255 255 255 / 0.08)',
+    },
+    '&:focus-visible': {
+      outline: '2px solid',
+      outlineOffset: '-2px',
+      outlineColor: (theme.vars || theme).palette.primary.main,
+    },
+  }),
+);
+
+const STORAGE_KEY = 'package-manager';
```

Comment:
Nice idea. Now it's feasible to do `<codeblock storageKey="package-manager">` which will save the value in the key `codeblock-package-manager`. I'm adding `codeblock-` to avoid naming collisions.